src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -2770,8 +2773,22 @@ class AccessibilityHelpWidget extends Widget implements ITerminalWidget {
 		}
 		content.push(this._descriptionForCommand(TerminalCommandId.OpenDetectedLink, openDetectedLink, openDetectedLinkNoKb));
 		content.push(readMore, dismiss);
-		const joinedContent = content.join('\n\n');
-		this._contentDomNode.domNode.appendChild(renderFormattedText(joinedContent));
-		this._contentDomNode.domNode.setAttribute('aria-label', joinedContent);
+		for (const co of content) {
+			const elt = renderFormattedText(co);
+			if (co === readMore) {
+				const link: HTMLElement = document.createElement('a');
+				elt.appendChild(link);
+				link.setAttribute('href', 'https://code.visualstudio.com/docs/editor/accessibility#_terminal-accessibility');
```

Comment:
Did you try `renderMarkdown` with `actionHandler`? https://github.com/microsoft/vscode/blob/81fbd20c41652eba0a33b417d12ab940cdf5f051/src/vs/workbench/contrib/preferences/browser/settingsTree.ts#L934-L955

Pulling the link and handling it specially like this feels like a code smell as it handles links differently to everywhere else. For example it doesn't have a click handler so it only works via keyboard.

Reply:
even if that works, markdown links aren't discoverable by a screen reader on macOS, so we cannot use that

Reply:
actually, it is accessible via tab in the markdown preview, so i'll try that

Reply:
not working (doesn't tab to the element and also doesn't respond to hitting enter).

Reply:
only responds to being clicked

Reply:
didn't work re tab index or enter, but I figured out a way around that